### ISSUE TITLE
Code to replace path separators on DOS-based filesystems is invalid

### ISSUE THREAD

**Issue by CohenArthur** [Rust-GCC/gccrs#2971]
```c++
  // On windows, the path might mix '/' and '\' separators. Replace the
  // UNIX-like separators by MSDOS separators to make sure the path will resolve
  // properly.
  //
  // Source: rustc compiler
  // (https://github.com/rust-lang/rust/blob/9863bf51a52b8e61bcad312f81b5193d53099f9f/compiler/rustc_expand/src/module.rs#L174)
#if defined(HAVE_DOS_BASED_FILE_SYSTEM)
  path.replace ('/', '\\');
#endif /* HAVE_DOS_BASED_FILE_SYSTEM */
```

this code is invalid, and throws an error about a lack of matching function. https://en.cppreference.com/w/cpp/string/basic_string/replace shows that indeed this is not a correct usage of the function, and that we should change it.

**Comment by CohenArthur** [comment]
This is also mentioned in #2843 and can be replaced with a call to `std::replace`

**Comment by CohenArthur** [comment]
we should have but I forgot :grimacing: I'll try and take care of it

**Comment by P-E-P** [comment]
Fixed in #3164
